packages/aws-cdk-lib/aws-ec2/lib/prefix-list.ts

```
@@ -76,21 +78,62 @@ abstract class PrefixListBase extends Resource implements IPrefixList {
   public abstract readonly prefixListId: string;
 }
 
+/**
+ * Properties for looking up an existing managed prefix list.
+ */
+export interface PrefixListLookupOptions {
+  /**
+   * The name of the prefix list to lookup.
+   */
+  readonly prefixListName: string;
+}
+
 /**
  * A managed prefix list.
  * @resource AWS::EC2::PrefixList
  */
 export class PrefixList extends PrefixListBase {
   /**
    * Look up prefix list by id.
-   *
    */
   public static fromPrefixListId(scope: Construct, id: string, prefixListId: string): IPrefixList {
     class Import extends Resource implements IPrefixList {
       public readonly prefixListId = prefixListId;
     }
     return new Import(scope, id);
   }
+
+  /**
+   * Look up prefix list by name
+   */
+  public static fromLookup(scope: Construct, id: string, options: PrefixListLookupOptions): IPrefixList {
+    if (Token.isUnresolved(options.prefixListName)) {
+      throw new ValidationError('All arguments to look up a managed prefix list must be concrete (no Tokens)', scope);
+    }
+
+    const dummyResponse = { PrefixListId: 'pl-xxxxxxxx' };
+    const response: {[key: string]: any}[] = ContextProvider.getValue(scope, {
+      provider: cxschema.ContextProvider.CC_API_PROVIDER,
+      props: {
+        typeName: 'AWS::EC2::PrefixList',
+        propertyMatch: {
+          PrefixListName: options.prefixListName,
+        },
+        propertiesToReturn: ['PrefixListId'],
+      } as Omit<cxschema.CcApiContextQuery, 'account'|'region'>,
+      dummyValue: [dummyResponse],
+    }).value;
+
+    // getValue returns a list of result objects. We are expecting 1 result or Error.
+    let prefixList = response[0];
+    if (!prefixList?.PrefixListId) {
+      Annotations.of(scope).addError(`Could not find the managed prefix list '${options.prefixListName}'`);
+      prefixList = dummyResponse;
```

for my understanding, why do we need the assignment here, can we just throw an error

---

Other `fromLookup()` methods never throws an error. An error thrown by context provider will be added as annotation in `ContextProvider.getValue()`.
https://github.com/aws/aws-cdk/blob/bff85e88fbe0ddb10bdcde8039af1736afa77cc2/packages/aws-cdk-lib/core/lib/context-provider.ts#L127-L131

Therefore, in my understanding, annotation seems to be preferred here.
If not, I can change to throw `ValidationError` here.

---

That's what I try to understand, why we want to add an error here, can we remove this?

---

If the managed prefix list with specified name does not exist, CC API context provider returns an empty array `[]`. We should detect it as an error.
Another solution: update the provider (in aws-cdk-lib) to return the dummy value with providerError if the result is empty.

---

Submitted #33826

---

Updated to throw ValidationError instead of adding an error annotation.
After the above PR is merged, we can assume exact one result.